core/src/main/java/com/linecorp/armeria/server/AggregatedHttpResponseHandler.java

```diff
@@ -68,18 +68,20 @@ public Void apply(@Nullable AggregatedHttpResponse response, @Nullable Throwable
 
     private void apply0(@Nullable AggregatedHttpResponse response, @Nullable Throwable cause) {
         clearTimeout();
+        if (failIfStreamOrSessionClosed()) {
```

Comment:
Question) Do you mind explaining why this change is necessary?

Reply:
If the session closed, we don't have to send the response using `recoverAndWrite(...)`. So we should check this first. 😉

Reply:
How about adding some comment?
```suggestion
        // Close early to avoid sending data unnecessarily on a closed stream.
        if (failIfStreamOrSessionClosed()) {
```